src/ImageDetector/index.js

```diff
@@ -0,0 +1,162 @@
+import * as tf from '@tensorflow/tfjs';
+import handleArguments from '../utils/handleArguments';
+import { mediaReady } from '../utils/imageUtilities';
+
+/**
+ * @typedef {Object} SpecificDetectorImplementation
+ *
+ * @property {Promise<boolean>} ready - lets the parent detector know that the
+ * specific implementation is ready.
+ *
+ * @property {(input: tf.Tensor3D) => Promise<Array>} detect - core detection method.
+ * the ImageDetector will call the `detect` method of the specific implementation.
+ * It should accept a TensorFlow tensor?? or an image??
+ * And return an array of detections??
```

Comment:
I was initially thinking that we would handle the image to tensor conversion here in the ImageDetector and call the specific model with the tensor.  But it seems like all of these models accept image or video and convert it themselves?